File: admin/ios/TigerAdmin/Sources/Models/AdminModels.swift
import Foundation

// MARK: - Admin User Models

struct AdminUser: Codable, Identifiable {
    let id: Int64
    let username: String
    let email: String
    let firstName: String?
    let lastName: String?
    let role: AdminRole
    let permissions: [String]
    let status: AdminStatus
    let twoFactorEnabled: Bool
    let lastLoginAt: String?
    let createdAt: String
    let updatedAt: String
    let avatarUrl: String?
    let phone: String?
    let department: String?
    
    var fullName: String {
        if let firstName = firstName, let lastName = lastName {
            return "\(firstName) \(lastName)"
        }
        return firstName ?? username
    }
}

enum AdminRole: String, Codable, CaseIterable {
    case superAdmin = "super_admin"
    case admin = "admin"
    case support = "support"
    case analyst = "analyst"
    case moderator = "moderator"
    
    var displayName: String {
        switch self {
        case .superAdmin: return "Super Admin"
        case .admin: return "Admin"
        case .support: return "Support"
        case .analyst: return "Analyst"
        case .moderator: return "Moderator"
        }
    }
    
    var permissions: [String] {
        switch self {
        case .superAdmin: return ["all"]
        case .admin: return ["users", "transactions", "kyc", "tokens", "fees", "analytics"]
        case .support: return ["users.view", "kyc.review"]
        case .analyst: return ["analytics.view", "reports.view"]
        case .moderator: return ["users.view", "transactions.view"]
        }
    }
}

enum AdminStatus: String, Codable {
    case active
    case suspended
    case inactive
}

// MARK: - Platform User Models

struct PlatformUser: Codable, Identifiable {
    let id: Int64
    let email: String
    let username: String?
    let walletAddress: String?
    let status: UserStatus
    let kycStatus: KYCStatus
    let kycLevel: Int
    let riskScore: Int
    let createdAt: String
    let lastLoginAt: String?
    let registrationIp: String?
    let tags: [String]
    let referredBy: String?
    let whiteLabelId: Int64?
}

enum UserStatus: String, Codable, CaseIterable {
    case active
    case pending
    case suspended
    case banned
    
    var displayName: String {
        rawValue.capitalized
    }
}

enum KYCStatus: String, Codable, CaseIterable {
    case none
    case pending
    case level1 = "level1"
    case level2 = "level2"
    case level3 = "level3"
    case rejected
    
    var displayName: String {
        switch self {
        case .none: return "Not Verified"
        case .pending: return "Pending"
        case .level1: return "Level 1"
        case .level2: return "Level 2"
        case .level3: return "Level 3"
        case .rejected: return "Rejected"
        }
    }
    
    var level: Int {
        switch self {
        case .none: return 0
        case .pending: return 0
        case .level1: return 1
        case .level2: return 2
        case .level3: return 3
        case .rejected: return 0
        }
    }
}

// MARK: - Transaction Models

struct Transaction: Codable, Identifiable {
    let id: Int64
    let hash: String
    let type: TransactionType
    let chain: String
    let fromAddress: String
    let toAddress: String
    let amount: String
    let token: String
    let tokenAmount: String?
    let status: TransactionStatus
    let blockNumber: Int64?
    let gasUsed: String?
    let gasPrice: String?
    let timestamp: String
    let flagged: Bool
    let flagReason: String?
    let userId: Int64
    
    var shortHash: String {
        let hash = self.hash
        if hash.count > 16 {
            return String(hash.prefix(8)) + "..." + String(hash.suffix(8))
        }
        return hash
    }
}

enum TransactionType: String, Codable, CaseIterable {
    case transfer
    case swap
    case stake
    case unstake
    case bridge
    case withdraw
    case deposit
    case mint
    case burn
    
    var displayName: String {
        rawValue.capitalized
    }
    
    var icon: String {
        switch self {
        case .transfer: return "arrow.left.arrow.right"
        case .swap: return "arrow.triangle.2.circlepath"
        case .stake: return "lock.fill"
        case .unstake: return "lock.open.fill"
        case .bridge: return "link"
        case .withdraw: return "arrow.down.circle"
        case .deposit: return "arrow.up.circle"
        case .mint: return "plus.circle"
        case .burn: return "minus.circle"
        }
    }
}

enum TransactionStatus: String, Codable, CaseIterable {
    case pending
    case confirmed
    case failed
    
    var displayName: String {
        rawValue.capitalized
    }
    
    var color: String {
        switch self {
        case .pending: return "orange"
        case .confirmed: return "green"
        case .failed: return "red"
        }
    }
}

// MARK: - KYC Models

struct KYCApplication: Codable, Identifiable {
    let id: Int64
    let userId: Int64
    let userEmail: String
    let level: Int
    let status: KYCApplicationStatus
    let submittedAt: String
    let reviewedAt: String?
    let reviewedBy: String?
    let rejectionReason: String?
    let documents: [KYCDocument]
    let ipAddress: String?
    let notes: String?
}

enum KYCApplicationStatus: String, Codable, CaseIterable {
    case pending
    case approved
    case rejected
    
    var displayName: String {
        rawValue.capitalized
    }
    
    var color: String {
        switch self {
        case .pending: return "orange"
        case .approved: return "green"
        case .rejected: return "red"
        }
    }
}

struct KYCDocument: Codable {
    let type: String
    let url: String
    let status: String
    let verifiedAt: String?
}

// MARK: - Token Models

struct Token: Codable, Identifiable {
    let id: Int64
    let name: String
    let symbol: String
    let contractAddress: String
    let chain: String
    let decimals: Int
    let totalSupply: String
    let logoUrl: String?
    let website: String?
    let description: String?
    let price: String?
    let marketCap: String?
    let volume24h: String?
    let priceChange24h: String?
    let isActive: Bool
    let isVerified: Bool
    let listingFee: String?
    let listedAt: String?
}

struct TokenListingRequest: Codable, Identifiable {
    let id: Int64
    let tokenSymbol: String
    let tokenName: String
    let contractAddress: String
    let chainId: Int64
    let tier: ListingTier
    let status: ListingStatus
    let requesterAddress: String
    let requesterEmail: String
    let oneTimeFee: String
    let monthlyFee: String
    let requestedAt: String
}

enum ListingTier: String, Codable {
    case basic
    case standard
    case premium
    case premiumPlus = "premium_plus"
    
    var displayName: String {
        switch self {
        case .basic: return "Basic"
        case .standard: return "Standard"
        case .premium: return "Premium"
        case .premiumPlus: return "Premium Plus"
        }
    }
}

enum ListingStatus: String, Codable, CaseIterable {
    case pending
    case approved
    case rejected
    
    var displayName: String {
        rawValue.capitalized
    }
}

// MARK: - Withdrawal Models

struct WithdrawalRequest: Codable, Identifiable {
    let id: Int64
    let userId: Int64
    let userEmail: String
    let amount: String
    let token: String
    let chain: String
    let toAddress: String
    let status: WithdrawalStatus
    let approvedAt: String?
    let approvedBy: String?
    let rejectedAt: String?
    let rejectionReason: String?
    let processedAt: String?
    let txHash: String?
    let fee: String?
    let createdAt: String
}

enum WithdrawalStatus: String, Codable, CaseIterable {
    case pending
    case approved
    case rejected
    case processing
    case completed
    case failed
    
    var displayName: String {
        rawValue.capitalized
    }
    
    var color: String {
        switch self {
        case .pending: return "orange"
        case .approved: return "blue"
        case .rejected: return "red"
        case .processing: return "purple"
        case .completed: return "green"
        case .failed: return "red"
        }
    }
}

// MARK: - White Label Models

struct WhiteLabel: Codable, Identifiable {
    let id: Int64
    let name: String
    let slug: String
    let domain: String?
    let logoUrl: String?
    let faviconUrl: String?
    let primaryColor: String
    let secondaryColor: String?
    let status: WhiteLabelStatus
    let contactEmail: String?
    let contactPhone: String?
    let address: String?
    let description: String?
    let features: [String]
    let feeStructure: FeeStructure?
    let createdAt: String
    let expiresAt: String?
}

enum WhiteLabelStatus: String, Codable {
    case active
    case suspended
    case pending
    
    var displayName: String {
        rawValue.capitalized
    }
}

struct FeeStructure: Codable {
    let tradingFee: String
    let withdrawalFee: String
    let depositFee: String
    let listingFee: String
}

// MARK: - Fee Configuration Models

struct FeeConfig: Codable, Identifiable {
    let id: Int64
    let feeType: String
    let chainId: Int64?
    let tokenSymbol: String?
    let feeAmountUsd: String
    let feePercentage: String
    let minFeeUsd: String
    let maxFeeUsd: String?
    let isActive: Bool
}

// MARK: - System Models

struct SystemStatus: Codable {
    let serviceName: String
    let status: String
    let uptime: String
    let latency: String
    let lastCheck: String
    
    var isHealthy: Bool {
        status == "running" || status == "healthy"
    }
}

struct SystemHealth: Codable {
    let status: String
    let uptime: String
    let memoryUsage: String
    let cpuUsage: String
    let diskUsage: String
}

// MARK: - Analytics Models

struct AnalyticsData: Codable {
    let totalUsers: Int64
    let activeUsers: Int64
    let totalVolume: String
    let dailyTransactions: Int64
    let totalFees: String
    let pendingKyc: Int64
    let systemHealth: String
    let timestamp: String
}

// MARK: - Bot Models

struct BotInstance: Codable, Identifiable {
    let id: Int64
    let userId: Int64
    let userEmail: String
    let botType: String
    let name: String
    let status: BotStatus
    let connectedDexs: Int
    let connectedCexs: Int
    let totalPnl: String
    let totalVolume: String
    let totalOrders: Int64
    let avgLatencyUs: Int64
    let createdAt: String
    let lastTradeAt: String?
}

enum BotStatus: String, Codable {
    case running
    case stopped
    case error
    case paused
    
    var displayName: String {
        rawValue.capitalized
    }
    
    var color: String {
        switch self {
        case .running: return "green"
        case .stopped: return "gray"
        case .error: return "red"
        case .paused: return "orange"
        }
    }
}

// MARK: - API Key Models

struct APIKey: Codable, Identifiable {
    let id: Int64
    let name: String
    let key: String
    let adminId: Int64
    let permissions: APIKeyPermissions
    let rateLimit: Int
    let lastUsedAt: String?
    let expiresAt: String?
    let status: String
    let createdAt: String
}

struct APIKeyPermissions: Codable {
    let trading: Bool
    let reading: Bool
    let withdrawal: Bool
}

// MARK: - Blockchain Models

struct Blockchain: Codable, Identifiable {
    let id: String
    let name: String
    let symbol: String
    let chainId: Int64
    let chainIdHex: String?
    let isEvm: Bool
    let isActive: Bool
    let explorerUrl: String?
    let rpcUrl: String?
    let nativeTokenSymbol: String
    let avgGasPriceGwei: Double
}

// MARK: - Pagination

struct PaginatedResponse<T: Codable>: Codable {
    let data: [T]
    let pagination: Pagination
}

struct Pagination: Codable {
    let page: Int
    let limit: Int
    let total: Int64
    let totalPages: Int
    
    var hasNextPage: Bool {
        page < totalPages
    }
    
    var hasPreviousPage: Bool {
        page > 1
    }
}

// MARK: - Notification Models

struct AdminNotification: Codable, Identifiable {
    let id: Int64
    let title: String
    let message: String
    let type: String
    let read: Bool
    let createdAt: String
    let data: [String: String]?
}

File: admin/android/app/src/main/java/com/tigeradmin/data/model/AdminModels.kt
package com.tigeradmin.data.model

import com.google.gson.annotations.SerializedName

/**
 * Admin User Model
 * Represents an admin user in the system
 */
data class AdminUser(
    @SerializedName("id")
    val id: Long,
    
    @SerializedName("username")
    val username: String,
    
    @SerializedName("email")
    val email: String,
    
    @SerializedName("first_name")
    val firstName: String?,
    
    @SerializedName("last_name")
    val lastName: String?,
    
    @SerializedName("role")
    val role: AdminRole,
    
    @SerializedName("permissions")
    val permissions: List<String>,
    
    @SerializedName("status")
    val status: AdminStatus,
    
    @SerializedName("two_factor_enabled")
    val twoFactorEnabled: Boolean,
    
    @SerializedName("last_login_at")
    val lastLoginAt: String?,
    
    @SerializedName("created_at")
    val createdAt: String,
    
    @SerializedName("updated_at")
    val updatedAt: String,
    
    @SerializedName("avatar_url")
    val avatarUrl: String?,
    
    @SerializedName("phone")
    val phone: String?,
    
    @SerializedName("department")
    val department: String?
)

/**
 * Admin Role Enum
 */
enum class AdminRole {
    @SerializedName("super_admin")
    SUPER_ADMIN,
    
    @SerializedName("admin")
    ADMIN,
    
    @SerializedName("support")
    SUPPORT,
    
    @SerializedName("analyst")
    ANALYST,
    
    @SerializedName("moderator")
    MODERATOR
}

/**
 * Admin Status Enum
 */
enum class AdminStatus {
    @SerializedName("active")
    ACTIVE,
    
    @SerializedName("suspended")
    SUSPENDED,
    
    @SerializedName("inactive")
    INACTIVE
}

/**
 * Platform User Model
 * Represents a user in the TigerWallet platform
 */
data class PlatformUser(
    @SerializedName("id")
    val id: Long,
    
    @SerializedName("email")
    val email: String,
    
    @SerializedName("username")
    val username: String?,
    
    @SerializedName("wallet_address")
    val walletAddress: String?,
    
    @SerializedName("status")
    val status: UserStatus,
    
    @SerializedName("kyc_status")
    val kycStatus: KYCStatus,
    
    @SerializedName("kyc_level")
    val kycLevel: Int,
    
    @SerializedName("risk_score")
    val riskScore: Int,
    
    @SerializedName("created_at")
    val createdAt: String,
    
    @SerializedName("last_login_at")
    val lastLoginAt: String?,
    
    @SerializedName("registration_ip")
    val registrationIp: String?,
    
    @SerializedName("tags")
    val tags: List<String>,
    
    @SerializedName("referred_by")
    val referredBy: String?,
    
    @SerializedName("white_label_id")
    val whiteLabelId: Long?
)

/**
 * User Status Enum
 */
enum class UserStatus {
    @SerializedName("active")
    ACTIVE,
    
    @SerializedName("pending")
    PENDING,
    
    @SerializedName("suspended")
    SUSPENDED,
    
    @SerializedName("banned")
    BANNED
}

/**
 * KYC Status Enum
 */
enum class KYCStatus {
    @SerializedName("none")
    NONE,
    
    @SerializedName("pending")
    PENDING,
    
    @SerializedName("level1")
    LEVEL1,
    
    @SerializedName("level2")
    LEVEL2,
    
    @SerializedName("level3")
    LEVEL3,
    
    @SerializedName("rejected")
    REJECTED
}

/**
 * Transaction Model
 */
data class Transaction(
    @SerializedName("id")
    val id: Long,
    
    @SerializedName("hash")
    val hash: String,
    
    @SerializedName("type")
    val type: TransactionType,
    
    @SerializedName("chain")
    val chain: String,
    
    @SerializedName("from_address")
    val fromAddress: String,
    
    @SerializedName("to_address")
    val toAddress: String,
    
    @SerializedName("amount")
    val amount: String,
    
    @SerializedName("token")
    val token: String,
    
    @SerializedName("token_amount")
    val tokenAmount: String?,
    
    @SerializedName("status")
    val status: TransactionStatus,
    
    @SerializedName("block_number")
    val blockNumber: Long?,
    
    @SerializedName("gas_used")
    val gasUsed: String?,
    
    @SerializedName("gas_price")
    val gasPrice: String?,
    
    @SerializedName("timestamp")
    val timestamp: String,
    
    @SerializedName("flagged")
    val flagged: Boolean,
    
    @SerializedName("flag_reason")
    val flagReason: String?,
    
    @SerializedName("user_id")
    val userId: Long
)

enum class TransactionType {
    @SerializedName("transfer") TRANSFER,
    @SerializedName("swap") SWAP,
    @SerializedName("stake") STAKE,
    @SerializedName("unstake") UNSTAKE,
    @SerializedName("bridge") BRIDGE,
    @SerializedName("withdraw") WITHDRAW,
    @SerializedName("deposit") DEPOSIT,
    @SerializedName("mint") MINT,
    @SerializedName("burn") BURN
}

enum class TransactionStatus {
    @SerializedName("pending") PENDING,
    @SerializedName("confirmed") CONFIRMED,
    @SerializedName("failed") FAILED
}

/**
 * KYC Application Model
 */
data class KYCApplication(
    @SerializedName("id") val id: Long,
    @SerializedName("user_id") val userId: Long,
    @SerializedName("user_email") val userEmail: String,
    @SerializedName("level") val level: Int,
    @SerializedName("status") val status: KYCApplicationStatus,
    @SerializedName("submitted_at") val submittedAt: String,
    @SerializedName("reviewed_at") val reviewedAt: String?,
    @SerializedName("reviewed_by") val reviewedBy: String?,
    @SerializedName("rejection_reason") val rejectionReason: String?,
    @SerializedName("documents") val documents: List<KYCDocument>,
    @SerializedName("ip_address") val ipAddress: String?,
    @SerializedName("notes") val notes: String?
)

enum class KYCApplicationStatus {
    @SerializedName("pending") PENDING,
    @SerializedName("approved") APPROVED,
    @SerializedName("rejected") REJECTED
}

data class KYCDocument(
    @SerializedName("type") val type: String,
    @SerializedName("url") val url: String,
    @SerializedName("status") val status: String,
    @SerializedName("verified_at") val verifiedAt: String?
)

/**
 * Token Model
 */
data class Token(
    @SerializedName("id") val id: Long,
    @SerializedName("name") val name: String,
    @SerializedName("symbol") val symbol: String,
    @SerializedName("contract_address") val contractAddress: String,
    @SerializedName("chain") val chain: String,
    @SerializedName("decimals") val decimals: Int,
    @SerializedName("total_supply") val totalSupply: String,
    @SerializedName("logo_url") val logoUrl: String?,
    @SerializedName("website") val website: String?,
    @SerializedName("description") val description: String?,
    @SerializedName("price") val price: String?,
    @SerializedName("market_cap") val marketCap: String?,
    @SerializedName("volume_24h") val volume24h: String?,
    @SerializedName("price_change_24h") val priceChange24h: String?,
    @SerializedName("is_active") val isActive: Boolean,
    @SerializedName("is_verified") val isVerified: Boolean,
    @SerializedName("listing_fee") val listingFee: String?,
    @SerializedName("listed_at") val listedAt: String?
)

/**
 * Token Listing Request Model
 */
data class TokenListingRequest(
    @SerializedName("id") val id: Long,
    @SerializedName("token_symbol") val tokenSymbol: String,
    @SerializedName("token_name") val tokenName: String,
    @SerializedName("contract_address") val contractAddress: String,
    @SerializedName("chain_id") val chainId: Long,
    @SerializedName("tier") val tier: ListingTier,
    @SerializedName("status") val status: ListingStatus,
    @SerializedName("requester_address") val requesterAddress: String,
    @SerializedName("requester_email") val requesterEmail: String,
    @SerializedName("one_time_fee") val oneTimeFee: String,
    @SerializedName("monthly_fee") val monthlyFee: String,
    @SerializedName("requested_at") val requestedAt: String
)

enum class ListingTier {
    @SerializedName("basic") BASIC,
    @SerializedName("standard") STANDARD,
    @SerializedName("premium") PREMIUM,
    @SerializedName("premium_plus") PREMIUM_PLUS
}

enum class ListingStatus {
    @SerializedName("pending") PENDING,
    @SerializedName("approved") APPROVED,
    @SerializedName("rejected") REJECTED
}

/**
 * Withdrawal Request Model
 */
data class WithdrawalRequest(
    @SerializedName("id") val id: Long,
    @SerializedName("user_id") val userId: Long,
    @SerializedName("user_email") val userEmail: String,
    @SerializedName("amount") val amount: String,
    @SerializedName("token") val token: String,
    @SerializedName("chain") val chain: String,
    @SerializedName("to_address") val toAddress: String,
    @SerializedName("status") val status: WithdrawalStatus,
    @SerializedName("approved_at") val approvedAt: String?,
    @SerializedName("approved_by") val approvedBy: String?,
    @SerializedName("rejected_at") val rejectedAt: String?,
    @SerializedName("rejection_reason") val rejectionReason: String?,
    @SerializedName("processed_at") val processedAt: String?,
    @SerializedName("tx_hash") val txHash: String?,
    @SerializedName("fee") val fee: String?,
    @SerializedName("created_at") val createdAt: String
)

enum class WithdrawalStatus {
    @SerializedName("pending") PENDING,
    @SerializedName("approved") APPROVED,
    @SerializedName("rejected") REJECTED,
    @SerializedName("processing") PROCESSING,
    @SerializedName("completed") COMPLETED,
    @SerializedName("failed") FAILED
}

/**
 * White Label Model
 */
data class WhiteLabel(
    @SerializedName("id") val id: Long,
    @SerializedName("name") val name: String,
    @SerializedName("slug") val slug: String,
    @SerializedName("domain") val domain: String?,
    @SerializedName("logo_url") val logoUrl: String?,
    @SerializedName("favicon_url") val faviconUrl: String?,
    @SerializedName("primary_color") val primaryColor: String,
    @SerializedName("secondary_color") val secondaryColor: String?,
    @SerializedName("status") val status: WhiteLabelStatus,
    @SerializedName("contact_email") val contactEmail: String?,
    @SerializedName("contact_phone") val contactPhone: String?,
    @SerializedName("address") val address: String?,
    @SerializedName("description") val description: String?,
    @SerializedName("features") val features: List<String>,
    @SerializedName("fee_structure") val feeStructure: FeeStructure?,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("expires_at") val expiresAt: String?
)

enum class WhiteLabelStatus {
    @SerializedName("active") ACTIVE,
    @SerializedName("suspended") SUSPENDED,
    @SerializedName("pending") PENDING
}

data class FeeStructure(
    @SerializedName("trading_fee") val tradingFee: String,
    @SerializedName("withdrawal_fee") val withdrawalFee: String,
    @SerializedName("deposit_fee") val depositFee: String,
    @SerializedName("listing_fee") val listingFee: String
)

/**
 * Fee Configuration Model
 */
data class FeeConfig(
    @SerializedName("id") val id: Long,
    @SerializedName("fee_type") val feeType: String,
    @SerializedName("chain_id") val chainId: Long?,
    @SerializedName("token_symbol") val tokenSymbol: String?,
    @SerializedName("fee_amount_usd") val feeAmountUsd: String,
    @SerializedName("fee_percentage") val feePercentage: String,
    @SerializedName("min_fee_usd") val minFeeUsd: String,
    @SerializedName("max_fee_usd") val maxFeeUsd: String?,
    @SerializedName("is_active") val isActive: Boolean
)

/**
 * System Status Model
 */
data class SystemStatus(
    @SerializedName("service_name") val serviceName: String,
    @SerializedName("status") val status: String,
    @SerializedName("uptime") val uptime: String,
    @SerializedName("latency") val latency: String,
    @SerializedName("last_check") val lastCheck: String
)

/**
 * Analytics Data Model
 */
data class AnalyticsData(
    @SerializedName("total_users") val totalUsers: Long,
    @SerializedName("active_users") val activeUsers: Long,
    @SerializedName("total_volume") val totalVolume: String,
    @SerializedName("daily_transactions") val dailyTransactions: Long,
    @SerializedName("total_fees") val totalFees: String,
    @SerializedName("pending_kyc") val pendingKyc: Long,
    @SerializedName("system_health") val systemHealth: String,
    @SerializedName("timestamp") val timestamp: String
)

/**
 * Bot Instance Model
 */
data class BotInstance(
    @SerializedName("id") val id: Long,
    @SerializedName("user_id") val userId: Long,
    @SerializedName("user_email") val userEmail: String,
    @SerializedName("bot_type") val botType: String,
    @SerializedName("name") val name: String,
    @SerializedName("status") val status: BotStatus,
    @SerializedName("connected_dexs") val connectedDexs: Int,
    @SerializedName("connected_cexs") val connectedCexs: Int,
    @SerializedName("total_pnl") val totalPnl: String,
    @SerializedName("total_volume") val totalVolume: String,
    @SerializedName("total_orders") val totalOrders: Long,
    @SerializedName("avg_latency_us") val avgLatencyUs: Long,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("last_trade_at") val lastTradeAt: String?
)

enum class BotStatus {
    @SerializedName("running") RUNNING,
    @SerializedName("stopped") STOPPED,
    @SerializedName("error") ERROR,
    @SerializedName("paused") PAUSED
}

/**
 * API Key Model
 */
data class APIKey(
    @SerializedName("id") val id: Long,
    @SerializedName("name") val name: String,
    @SerializedName("key") val key: String,
    @SerializedName("admin_id") val adminId: Long,
    @SerializedName("permissions") val permissions: APIKeyPermissions,
    @SerializedName("rate_limit") val rateLimit: Int,
    @SerializedName("last_used_at") val lastUsedAt: String?,
    @SerializedName("expires_at") val expiresAt: String?,
    @SerializedName("status") val status: String,
    @SerializedName("created_at") val createdAt: String
)

data class APIKeyPermissions(
    @SerializedName("trading") val trading: Boolean,
    @SerializedName("reading") val reading: Boolean,
    @SerializedName("withdrawal") val withdrawal: Boolean
)

/**
 * Blockchain Model
 */
data class Blockchain(
    @SerializedName("id") val id: String,
    @SerializedName("name") val name: String,
    @SerializedName("symbol") val symbol: String,
    @SerializedName("chain_id") val chainId: Long,
    @SerializedName("chain_id_hex") val chainIdHex: String?,
    @SerializedName("is_evm") val isEvm: Boolean,
    @SerializedName("is_active") val isActive: Boolean,
    @SerializedName("explorer_url") val explorerUrl: String?,
    @SerializedName("rpc_url") val rpcUrl: String?,
    @SerializedName("native_token_symbol") val nativeTokenSymbol: String,
    @SerializedName("avg_gas_price_gwei") val avgGasPriceGwei: Double
)
